src/chapters/interfaces-and-polymorphism/interfaces.rst

```diff
@@ -0,0 +1,294 @@
+
+Method Signatures
+^^^^^^^^^^^^^^^^^
+
+.. index:: ! method signature
+
+One really useful aspect of using an interface is the ability to specify method signatures.
+A **method signature** includes the name, parameters, and return
+type of a method, but no body.
+
+As part of our cat application, let's create a method signature, ``Eat``, as part of an 
+interface, ``IFeedable``.
+
+.. admonition:: Example
+
+   .. sourcecode:: c#
+      :linenos:
+
+      public interface IFeedable
+      {
+
+         void Eat();
+
+      }
+
+In the code above, notice the following:
+
+#. We need to use the ``interface`` keyword to define our interface, ``IFeedable``.
+#. ``Eat`` only has a signature. We are NOT allowed to provide a body for methods defined 
+   in interfaces.
+#. ``Eat`` also does not have an access modifier. Interface members are always ``public``, 
+   and while we may use the ``public`` modifier, it’s unnecessary. An interface method may NOT 
+   have an access modifier that is more restrictive than ``public``.
+#. The ``IFeedable`` interface itself is declared ``public``, which means any other class may 
+   use it. We may also leave off ``public``, making the interface ``protected internal``, or 
+   usable only within the same assembly. Recall this access modifier described in this
+   :ref:`table <access-modifier-table>`.
+#. The name is indicative of the behavior that the interface is intended to describe. While this 
+   is only a convention, most interfaces have names that are adjectives preceded with an "I". 
+   While this is only a convention, you should follow it in the interest of code-readability.
+
+Static Methods
+^^^^^^^^^^^^^^
+
+A static method in an interface can contain code in the body.
+However, a static method cannot contain any references to instance properties in other classes.
+This means that our static methods should only deal with universal behaviors that are NOT 
+dependent on instance properties.
+
+
+Default Methods
+^^^^^^^^^^^^^^^
+
+.. index:: ! default methods
+
+A **default method** has a body and is a fully-formed method. It may be extended by classes 
+implementing the interface.
+
+.. sourcecode:: c#
+   :linenos:
+
+   public interface IFeedable {
+
+      void Eat();
+
+      void Nap() {
+         Console.WriteLine("snooooozzze");
+      }
+
+   }
+
+The intended purpose of default methods is to allow
+programmers to add a method to an interface that has already been
+released, while not forcing those already using the interface to add new
+code to their classes. *You should avoid using default methods in all situations other than the 
+one described here.*
+
+Implementing an Interface
+-------------------------
+
+The purpose of an interface is to define a contract of behaviors that classes uphold. In doing so, we say that they “*implement* the
+interface”. The syntax for implementation is the same as that for
+inheritance. Here’s how we can use the ``IFeedable`` interface in
```

Comment:
Good spot to remind them why the naming convention is so important!

Reply:
i dont understand this note, can you explain what you mean here?

Reply:
ok fixed this from your note on slack. lmk what you think